specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/ipAllocation.json

```diff
@@ -0,0 +1,395 @@
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The operation returns a list of IpAllocation resources.",
+            "schema": {
+              "$ref": "#/definitions/IpAllocationListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List all IpAllocations": {
+            "$ref": "./examples/IpAllocationList.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/IpAllocations": {
+      "get": {
+        "tags": [
+          "IpAllocations"
+        ],
+        "operationId": "IpAllocations_ListByResourceGroup",
+        "description": "Gets all IpAllocations in a resource group.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The operation returns a list of IpAllocation resources.",
+            "schema": {
+              "$ref": "#/definitions/IpAllocationListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List IpAllocations in resource group": {
+            "$ref": "./examples/IpAllocationListByResourceGroup.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "IpAllocationPropertiesFormat": {
+      "properties": {
+        "subnet": {
+          "$ref": "./virtualNetwork.json#/definitions/Subnet",
+          "description": "The Subnet that using the prefix of this IpAllocation resource."
+        },
+        "virtualNetwork": {
+          "$ref": "./virtualNetwork.json#/definitions/VirtualNetwork",
+          "description": "The VirtualNetwork that using the prefix of this IpAllocation resource."
+        },
```

Comment:
This isn't in the examples.  Is this creating a subnet/vnet?  Why does it contain the full definition of these items?  If its just a read-only reference to them it should only be the resource ID (and be marked readonly in swagger)

Reply:
Fixed with reference SubResource.